test/fixtures/es-module-loaders/mock-loader.mjs

```diff
@@ -192,48 +59,38 @@ export async function resolve(specifier, context, defaultResolve) {
 }
 
 export async function load(url, context, defaultLoad) {
-  doDrainPort();
```

Comment:
I would expect that `doDrainPort` is still necessary if we don't want the test to be flaky

Reply:
It doesn't seem to be necessary.

Reply:
How can you tell?

Reply:
I tried running `./tools/test.py --repeat=9999 test/es-module/test-esm-loader-mock.mjs` on my local machine, got `[02:35|% 100|+ 9995|-   4]: Done`. So the new test _is_ flaky, albeit not by much. I have tested if it would remove the flakiness, but that's the first thing that comes to mind. Having to deal with a flaky CI is really a pita, let's not take any chance, wdyt?

Reply:
If it's flaky then the implementation itself is flaky. The other tests following this pattern will have the same issue. We can add drainPort to all of them or we can add that functionality to core.

Reply:
What’s the justification for removing it? If letting it here minimizes the diff and avoids introducing some flakiness, there must be some strong reason for removing it.

Reply:
Well look at the comment attached to `doDrainPort`:

```
/**
 * FIXME: this is a hack to workaround loaders being
 * single threaded for now, just ensures that the MessagePort drains
 */
```

I don’t mind adding it back, but it seems like this masks a bug. Do you have time to add it back on your local and run the `repeat=9999` again to see if this fixes the flakiness?

Reply:
I missed that comment, thanks for the pointer. So it sounds like we should have removed it earlier – or removed that comment if it’s not accurate 🤔 I’m not on my computer right now, will check later.

Reply:
I just ran `./tools/test.py --repeat=9999 --timeout=30 test/es-module/test-esm-loader-mock.mjs` twice on my machine after pushing the latest commit and I get zero failures.

Reply:
I tried:
- on `main` with `doDrainPort`: `[02:47|% 100|+ 9999|-   0]: Done `
- on `main` without `doDrainPort`: `[02:45|% 100|+ 9997|-   2]: Done`

It seems the added flakiness is real and related to `doDrainPort`. I can try with more runs if that helps, but the order of operations from one thread to the next is inherently unpredictable, so it makes sense to me that draining the port helps. I'd be in favor of keeping `doDrainPort` and update the comment. I can do that in a separate PR if you prefer.

Reply:
Can you try with the latest commit in this branch? It might be that the original implementation was flaky without drain port but this new one is stable.